migrations/migration.py

```diff
@@ -0,0 +1,231 @@
+"""Migration Module
+
+Captures common methods needed by the system for migration
+"""
+
+import os
+from typing import List
+from datetime import datetime
+import uuid
+import imp
+
+from migrations.migration_resource import MigrationManager
+from migrations.utils import Node, LinkedList
+from isacc_messaging.audit import audit_entry
+from isacc_messaging.models.fhir import (
+    first_in_bundle,
+)
+
+
+class Migration:
+    def __init__(self, migrations_dir=None):
+        if migrations_dir is None:
+            migrations_dir = os.path.join(os.path.dirname(__file__), "versions")
+        self.migrations_dir = migrations_dir
+        self.migration_sequence = LinkedList()
+        self.build_migration_sequence()
+
+    def build_migration_sequence(self):
+        migration_files: list = self.get_migration_files()
+        migration_nodes: dict = {}
+
+        for filename in migration_files:
+            migration: str = filename[:-3]
+            migration_nodes[migration] = self.get_previous_migration_id(migration)
+
+        if len(migration_files) > 0:
+            try:
+                self.migration_sequence.build_list_from_dictionary(migration_nodes)
+            except:
+                error_message = "Cycle detected in migration sequence"
+                audit_entry(error_message, level='error')
+                raise ValueError(error_message)
+        else:
+            error_message = "No valid migration files."
+            audit_entry(error_message, level='info')
+
+
+    def get_migration_files(self) -> list:
+        migration_files = os.listdir(self.migrations_dir)
+        python_files = [file for file in migration_files if file.endswith(".py")]
+        return python_files
+
+    def get_previous_migration_id(self, migration_id: str) -> str:
+        """Retrieve the down_revision from a migration script."""
+        filename = migration_id + ".py"
+        down_revision = None
+        migration_path = os.path.join(self.migrations_dir, filename)
+        if os.path.exists(migration_path):
+            try:
+                migration_module = imp.load_source("migration_module", migration_path)
+                down_revision = getattr(migration_module, "down_revision", None)
+            except Exception as e:
```

Comment:
catching base `Exception` is dangerous.  it will often lead to other unexpected situations being caught and treated like the expected case.

please except the type of exception you expect to catch only.

also, in this case, I would think you'd want to re `raise e` as it's rather exceptional to find an ill defined migration file - and without re-rasing, the situation could easily be ignored.